src/UniDeposit.sol

```diff
@@ -53,64 +55,68 @@ contract UniDeposit {
         _;
     }
 
+    function unminted_USDT() view public returns (uint) {
+        return IERC20(USDT).balanceOf(address(this));
+    }    
+    function minted_yCRV() view public returns (uint) {
+        return IERC20(yCrvToken).balanceOf(address(this));
+    }
     function get_yCrvFromUsdt(uint256 amount) public view returns (uint256) {
-        return amount.mul(minted_yCRV).div(mintedUSDT);
+        return amount.mul(minted_yCRV()).div(mintedUSDT);
     }
+    function get_usdtFromYcrv(uint256 amount) public view returns (uint256) {
+        return amount.mul(mintedUSDT).div(minted_yCRV());
+    }    
 
     event Deposit(address indexed who, uint256 amountOfUsdt);
 
-    function deposit(uint256 amount) public goodToChargeUSDT(amount) {
+    function deposit(uint256 input) external goodToChargeUSDT(input) {
         // sadly no return from USDT
-        IUSDT(USDT).transferFrom(msg.sender, address(this), amount);
-        if (amount > mintedUSDT) {
+        IUSDT(USDT).transferFrom(msg.sender, address(this), input);
+        if (input > mintedUSDT) {
             // New Deposit goes to unmintedUSDT pool
-            unmintedUSDT = unmintedUSDT.add(amount);
-            emit Deposit(msg.sender, amount);
-            setBalance(msg.sender, balanceOf(msg.sender).add(amount));
+            setBalance(msg.sender, balanceOf(msg.sender).add(input));
+            emit Deposit(msg.sender, input);
         } else {
             // if enough, just swap then
-            uint256 yCrvWillGet = get_yCrvFromUsdt(amount);
-            IERC20(yCrvToken).transfer(msg.sender, yCrvWillGet);
+            uint256 output = get_yCrvFromUsdt(input);
+            mintedUSDT = mintedUSDT.sub(input);            
+            IERC20(yCrvToken).transfer(msg.sender, output);
         }
     }
 
+    function withdraw(uint input) external {
```

Comment:
This one *is not* a bug though, but It makes me confused while I am reviewing.

I don't see this function have anything to do with `balance`.

So why this function was called `withdraw`?

If you take yCrv from `msg.sender` and send USDT back to him, shouldn't it called `swapToUsdt()` or something else?

Maybe we could have a `withdraw` function, for someone who deposited USDT already but needs to take back his USDT deposit before the next round of yCrv minting?

Reply:
Withdraw() is yyCrv to USDT without 3% P3D.